main.playground/Contents.swift

```diff
@@ -1,26 +1,121 @@
 class Fruit {
-    static var id = 0;
-    var name: String;
-    var weight: Float; // grams
-    var price: Float; // cents
-    var id: Int;
-    
-    init(name: String, weight: Float, price: Float) {
-        Fruit.id = Fruit.id + 1;
-        self.id = Fruit.id;
-        self.name = name;
-        self.weight = weight;
-        self.price = price;
+    
+    static var id: Int = 0
+    var id: Int
+    var type: String
+    // Weight in grams
+    var weight: Float
+    
+    init(weight: Float, type: String) {
+        Fruit.id = Fruit.id + 1
+        self.id = Fruit.id
+        self.type = type
+        self.weight = weight
+    }
+    
+    func toString() -> String {
+        return "ID: \(self.id), Type: \(self.type), Weight: \(self.weight)gr"
+    }
+}
+
+class Greengrocer {
+    
+    var fruits:[String: [Fruit]] = [:]
+    // var fruits2:[String: (Float, [Fruit])] = [:]
+    var prices:[String:Float] = [:]
+    
+    // Potremmo inizializzare la classe con il valore del prezzo/kg di ogni singolo frutto
+    // passando come parametro un dictionary come -prices-, ma per semplicità usiamo un approccio diverso
+    init(){}
+    
+    func addFruit(content: [Fruit], price: Float) {
+        self.fruits.updateValue(content, forKey: content[0].type)
+        self.prices.updateValue(price, forKey: content[0].type)
+    }
+    
+    func countFruitTypes() -> String{
+        var counter: Int = 0
+        var output: String = "Ci sono "
+        for (index, value) in self.fruits {
+            if counter == 0 {
+                output.append("\(value.count) \(index)s")
+                counter += 1
+            }
+            else if self.fruits.count  != counter + 1 {
+                output.append(", \(value.count) \(index)s")
+                counter += 1
+            }
+            else {
+                output.append(" e \(value.count) \(index)s")
+            }
+        }
+        return output
+    }
+    
+    func totalValue() -> Float {
+        var total: Float = 0
+        
+        for (index, value) in self.fruits {
+            for item in value {
+                if let key = self.prices[index] {
+                    total += item.weight / 100 * key
+                }
+            }
+        }
+        
+        return total
+    }
```

Comment:
Se inserisci il prezzo nella classe `Fruit`, in teoria ti dovrebbe bastare prendere il prezzo di un frutto e moltiplicarlo per il numero di items di quel tipo, presenti nell'array.
```suggestion
    func totalValue() -> Float {
        var total: Float = 0
        
        for (index, value) in self.fruits {
            total += value.count * value[0].price
        }
        return total
    }
```

Reply:
o comunque se ci si basa sul prezzo 
```suggestion
    func totalValue() -> Float {
        var total: Float = 0
        
        for (index, value) in self.fruits {
            for fruit in value {
               total += fruit.weight / 100 * fruit.price
            }
        }
        return total
    }
```